app/models/concerns/profile_section.rb

```diff
@@ -5,8 +5,9 @@ module ProfileSection
     def prepare(**, &block)
       find_or_initialize_by(**).tap do |record|
         if record.new_record?
-          if (previous_application = jobseeker(record).job_applications.last)
-            copy_attributes(record, previous_application)
+          # do not populated from draft applications as they make be incomplete and invalid.
+          if (previously_submitted_application = jobseeker(record).job_applications.where.not(status: :draft).last)
```

Comment:
I was going to say that we should use a scope here and found after_submission - except it has a bug (created by yours truly) which forgets to add 'interviewing' to the list of 'after submission' values. Think we could use this, and change the implementation to 'not draft'?

Reply:
rails 7 has a 'invert_where' for just this sort of thing